src/v3/src/transformer/i18n/transform.ts

```diff
@@ -10,75 +10,58 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { IdxTransaction, Input } from '@okta/okta-auth-js';
+import { IdxTransaction } from '@okta/okta-auth-js';
 
 import { getI18NParams, getI18NValue, getMessage } from '../../../../v2/ion/i18nTransformer';
-import { AUTHENTICATOR_KEY } from '../../constants';
 import {
+  AuthenticatorButtonElement,
   FieldElement,
   FormBag,
   TransformStepFnWithOptions, UISchemaElement,
 } from '../../types';
 import { loc } from '../../util';
-import { getOptionValue } from '../selectAuthenticator/utils';
+import { traverseLayout } from '../util';
 
-const geti18nPath = (fieldName: string, stepName: string, authenticatorKey: string): string => {
-  const authenticatorKeyPath = authenticatorKey
-    ? `.${authenticatorKey}`
-    : '';
-  return `${stepName}${authenticatorKeyPath}.${fieldName}`;
-};
-
-const updateLabel = (transaction: IdxTransaction, element: FieldElement): void => {
+const getI18nOptions = (transaction: IdxTransaction, element: UISchemaElement) => {
   const { nextStep } = transaction;
-  const { relatesTo, name: stepName = '' } = nextStep || {};
+  const { relatesTo, name: stepName = '' } = nextStep!;
   const authenticatorKey = relatesTo?.value?.key || '';
-  const i18nPath = geti18nPath(element.options.inputMeta.name, stepName, authenticatorKey);
+  const fieldName = (element as FieldElement).options?.inputMeta?.name;
 
-  if (element.label) {
-    const params = getI18NParams(nextStep, authenticatorKey);
-    // eslint-disable-next-line no-param-reassign
-    element.label = getI18NValue(i18nPath, element.label, params);
+  let path = '';
+  const params = getI18NParams(nextStep, authenticatorKey);
+  if (element.type === 'Field') {
+    path = authenticatorKey
+      ? `${stepName}.${authenticatorKey}.${fieldName}`
+      : `${stepName}.${fieldName}`;
+  } else if (element.type === 'AuthenticatorButton') {
+    const methodType = (
+      element as AuthenticatorButtonElement
+    ).options.authenticator?.methods?.[0]?.type;
+    path = methodType
+      ? `${stepName}.authenticator.${(element as AuthenticatorButtonElement).options.key}.${methodType}`
+      : `${stepName}.authenticator.${(element as AuthenticatorButtonElement).options.key}`;
```

Comment:
This is the part that causes the issue with the authenticator labels. As you can see in the map, only a few authenticators uses the method Type in the mapping (only OV custom_app), but the methodType is available on more than that from the ion response. 

I think this should be updated to fix that so we wont have to go back and research why it doesn't work later (it should be a fairly straightforward fix).

https://github.com/okta/okta-signin-widget/blob/master/src/v2/ion/i18nTransformer.js#L66-L91

Reply:
will be handled in my next PR, still we will need tests, otherwise it can regress at any time

Reply:
Agree, there is already an ok-pl test in v2 that I thought we would re-use. I tried loading the language in jest but it didnt seem to work.

Reply:
don't worry about that, there will be a generic in jest layer to run against all mocks (only against transformation layer), but not sure when will it be added